src/test/java/org/junit/internal/runners/statements/ExpectExceptionTest.java

```diff
@@ -0,0 +1,29 @@
+package org.junit.internal.runners.statements;
+
+import org.junit.Test;
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.runners.model.Statement;
+
+import static org.junit.Assert.*;
+
+public class ExpectExceptionTest {
+
+    @Test
+    public void givenWeAreExpectingAssumptionViolatedExceptionAndAStatementThrowingAssumptionViolatedException() {
+        ExpectException sut = new ExpectException(new StatementThrowingAssumptionViolatedException(), AssumptionViolatedException.class);
```

Comment:
Also, if you are adding unit tests, please add a test that makes sure that tests that expect `AssumptionViolatedException` but do not throw it should fail